src/conditions/Condition.js

```diff
@@ -0,0 +1,60 @@
+const Joi = require('joi')
+
+class Condition {
+
+  static get SCHEMA() {
+    return Joi.object({
+      operator: Joi.string().allow(null).default(null),
+      left: Joi.alternatives().try(
+        Joi.number().strict(),
+        Joi.string(),
+        Joi.boolean(),
+        Joi.link('...'),
```

Comment:
allows recursive schema ... this means: `left` can be the same schema as the wrapping object (`condition`)